docs/source/migration-guides/0.15-deserialization.md

```diff
@@ -0,0 +1,292 @@
+
+Note that the `QueryResult::rows` field is not available anymore. If you used to access it directly, you need to change your code to use the helper methods instead.
+
+Before:
+
+```rust
+# extern crate scylla;
+# use scylla::LegacySession;
+# use std::error::Error;
+# async fn check_only_compiles(session: &LegacySession) -> Result<(), Box<dyn Error>> {
+let iter = session
+    .query_unpaged("SELECT name, age FROM my_keyspace.people", &[])
+    .await?
+    .rows_typed::<(String, i32)>()?;
+for row in iter {
+    let (name, age) = row?;
+    println!("{} has age {}", name, age);
+}
+# Ok(())
+# }
+```
+
+After:
+
+```rust
+# extern crate scylla;
+# use scylla::Session;
+# use std::error::Error;
+# async fn check_only_compiles(session: &Session) -> Result<(), Box<dyn Error>> {
+// 1. Note that the result must be converted to a rows result, and only then
+// an iterator created.
+let result = session
+    .query_unpaged("SELECT name, age FROM my_keyspace.people", &[])
+    .await?
+    .into_rows_result()?
+    .unwrap();
+
+// 2. Note that `rows` is used here, not `rows_typed`.
+// 3. Note that the new deserialization framework support deserializing types
+//    that borrow directly from the result frame; let's use them to avoid
+//    needless allocations.
+for row in result.rows::<(&str, i32)>()? {
+    let (name, age) = row?;
+    println!("{} has age {}", name, age);
+}
+# Ok(())
+# }
+```
+
+### Iterator queries
+
+The `Session::query_iter` and `Session::execute_iter` have been adjusted, too. They now return a `QueryPager` - an intermediate object which needs to be converted into `TypedRowStream` first before being actually iterated over.
+
+Before:
+
+```rust
+# extern crate scylla;
+# extern crate futures;
+# use scylla::LegacySession;
+# use std::error::Error;
+# use scylla::IntoTypedRows;
+# use futures::stream::StreamExt;
+# async fn check_only_compiles(session: &LegacySession) -> Result<(), Box<dyn Error>> {
+let mut rows_stream = session
+    .query_iter("SELECT name, age FROM my_keyspace.people", &[])
+    .await?
+    .into_typed::<(String, i32)>();
+
+while let Some(next_row_res) = rows_stream.next().await {
+    let (a, b): (String, i32) = next_row_res?;
+    println!("a, b: {}, {}", a, b);
+}
+# Ok(())
+# }
+```
+
+After:
+
+```rust
+# extern crate scylla;
+# extern crate futures;
+# use scylla::Session;
+# use std::error::Error;
+# use futures::stream::StreamExt;
+# async fn check_only_compiles(session: &Session) -> Result<(), Box<dyn Error>> {
+let mut rows_stream = session
+    .query_iter("SELECT name, age FROM my_keyspace.people", &[])
+    .await?
+    .rows_stream::<(String, i32)>()?;
+
+while let Some(next_row_res) = rows_stream.next().await {
+    let (a, b): (String, i32) = next_row_res?;
+    println!("a, b: {}, {}", a, b);
+}
```

Comment:
Can you add a comment about stream API not supporting borrowed types? Here, or even better above this snippet.

Question: Why is type annotation needed in `let (a, b): (String, i32) = next_row_res?;`? You already provided the type in `rows_stream` call. Is it to make the code easier to read or is it actually required for some reason?

Reply:
I don't, I just wanted to know why it is there. Type annotation for clarity are good.

Reply:
I removed it, and added a comment about possible use of turbofish.